### PR TITLE
MES-4142-Fixed fault count calculation for sub cat Ds

### PR DESCRIPTION
**Pull request by amobarak**
## Description

Check and update WRTC page for cat Ds, fixed fault count calculation for DE and D1E.

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts [hunk 1]
```diff
@@ -210,8 +180,8 @@ export class FaultCountDHelper {
 
     const seriousFaultSumOfSimpleCompetencies = Object.keys(pickBy(seriousFaults)).length;
     const vehicleCheckSeriousFaults =
-      vehicleChecks ? FaultCountDHelper.getVehicleChecksFaultCountCatD(vehicleChecks).seriousFaults : 0;
-    const pcvDoorExerciseFaultCount: number = get(pcvDoorExercise, 'seriousFault') ? 1 : 0;
+      vehicleChecks ? FaultCountDHelper.getVehicleChecksFaultCountNonTrailer(vehicleChecks).seriousFaults : 0;
+    const pcvDoorExerciseFaultCount: number = get(pcvDoorExercise, 'dangerousFault') ? 1 : 0;
```
Comment (rossfauldsbjss) [inline src/providers/fault-count/cat-d/fault-count.cat-d.ts]: this  is calculating serious faults yet the door exercise is looking at dangerousFault, surely that's wrong?

**Review by rossfauldsbjss — CHANGES_REQUESTED** [review]
Needs to look at serious fault for pcv exercise instead of dangerous when adding up serious faults

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts [hunk 2]
```diff
@@ -288,7 +258,31 @@ export class FaultCountDHelper {
     return result;
   }
 
-  private static getSafetyQuestionsFaultCount = (
+  public static getVehicleChecksFaultCountCatD = (
```
Comment (TomBillingtonUK) [inline src/providers/fault-count/cat-d/fault-count.cat-d.ts]: Convention is that public methods should be at the top of the file followed by protected and then private, can you move this function and the one below please.

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts (hunk 1 above)
Comment (amobarak) [reply src/providers/fault-count/cat-d/fault-count.cat-d.ts]: Very good spot Ross !!

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts (hunk 2 above)
Comment (amobarak) [reply src/providers/fault-count/cat-d/fault-count.cat-d.ts]: Thanks, will do

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts [hunk 3]
```diff
@@ -170,8 +182,8 @@ export class FaultCountDHelper {
     const result =
       faultTotal +
       sumManoeuvreFaults(manoeuvres, CompetencyOutcome.DF) +
-      pcvDoorExerciseFaultCount +
-      FaultCountDHelper.getVehicleChecksFaultCountCatD(vehicleChecks).drivingFaults;
+      FaultCountDHelper.getVehicleChecksFaultCountNonTrailer(vehicleChecks).drivingFaults +
```
Comment (Cavalier07) [inline src/providers/fault-count/cat-d/fault-count.cat-d.ts]: should be:

`this.getVehicleChecksFaultCountNonTrailer(vehicleChecks).drivingFaults`

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts [hunk 4]
```diff
@@ -193,9 +205,9 @@ export class FaultCountDHelper {
     const result =
       faultTotal +
       sumManoeuvreFaults(manoeuvres, CompetencyOutcome.DF) +
-      FaultCountDHelper.getVehicleChecksFaultCountCatD(vehicleChecks).drivingFaults +
-      pcvDoorExerciseFaultCount +
-      uncoupleRecoupleHasDrivingFault;
+      FaultCountDHelper.getVehicleChecksFaultCountTrailer(vehicleChecks).drivingFaults +
```
Comment (Cavalier07) [inline src/providers/fault-count/cat-d/fault-count.cat-d.ts]: should be:

`this.getVehicleChecksFaultCountTrailer(vehicleChecks).drivingFaults`

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts [hunk 5]
```diff
@@ -210,7 +222,7 @@ export class FaultCountDHelper {
 
     const seriousFaultSumOfSimpleCompetencies = Object.keys(pickBy(seriousFaults)).length;
     const vehicleCheckSeriousFaults =
-      vehicleChecks ? FaultCountDHelper.getVehicleChecksFaultCountCatD(vehicleChecks).seriousFaults : 0;
+      vehicleChecks ? FaultCountDHelper.getVehicleChecksFaultCountNonTrailer(vehicleChecks).seriousFaults : 0;
```
Comment (Cavalier07) [inline src/providers/fault-count/cat-d/fault-count.cat-d.ts]: should be:

`this.getVehicleChecksFaultCountNonTrailer(vehicleChecks).seriousFaults`

File: src/providers/fault-count/cat-d/fault-count.cat-d.ts [hunk 6]
```diff
@@ -232,7 +244,7 @@ export class FaultCountDHelper {
 
     const seriousFaultSumOfSimpleCompetencies = Object.keys(pickBy(seriousFaults)).length;
     const vehicleCheckSeriousFaults =
-      vehicleChecks ? FaultCountDHelper.getVehicleChecksFaultCountCatD(vehicleChecks).seriousFaults : 0;
+      vehicleChecks ? FaultCountDHelper.getVehicleChecksFaultCountTrailer(vehicleChecks).seriousFaults : 0;
```
Comment (Cavalier07) [inline src/providers/fault-count/cat-d/fault-count.cat-d.ts]: should be:

`this.getVehicleChecksFaultCountTrailer(vehicleChecks).seriousFaults`

**Review by rossfauldsbjss — APPROVED** [review]
looks good

**Review by TomBillingtonUK — APPROVED** [review]

**Review by faggc — APPROVED** [review]

**Pull request merged**